### PR TITLE
Add optional priority feature

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by vchuravy** [comment]
@luraess is this roughly what you had in mind? To give the communication task a higher priority?

File: examples/mpi.jl [hunk 1]
```diff
@@ -21,7 +21,8 @@ end
 
 function exchange!(h_send_buf, d_recv_buf, h_recv_buf, src_rank, dst_rank, comm)
     recv_req = MPI.Irecv!(h_recv_buf, src_rank, 666, comm)
-    recv = Base.Threads.@spawn begin 
+    recv = Base.Threads.@spawn begin
+        KernelAbstractions.priority!(backend, :high)
         cooperative_test!(recv_req)
         KernelAbstractions.copyto!(backend, d_recv_buf, h_recv_buf)
         KernelAbstractions.synchronize(backend) # Gurantueed to be cooperative
```
Comment (luraess) [inline examples/mpi.jl:28]:
```suggestion
        KernelAbstractions.synchronize(backend) # Guaranteed to be cooperative
```

**Comment by luraess** [comment]
Yeah - that's the direction. How is it implemented now? Do you have a pool of queues / streams per priority?

**Comment by vchuravy** [comment]
> Do you have a pool of queues / streams per priority?

No this would change the priority of the stream owned by the Task you are running on.

**Comment by luraess** [comment]
> No this would change the priority of the stream owned by the Task you are running on.

True - need to get used to the task based approach. Now with priorities, can one then implement a workflow to have one kernel to execute parts of the computations on the high priority queue / stream, have the MPI comm on that same task (I guess), and compute inner points on another low priority stream (asynchronously on another task)?

**Pull request merged**